
Epigenetics, as shown in the introduction (section \ref{sec:bioback}), is a pretty wide and complex field, and the sequencing technology to adopt depends on the biological question under investigation.

Some studies \cite{Koberstein2018, Auerbach2009} have demonstrated the importance of genome-wide chromatin accessibility of a broad spectrum of chromatin phenomena activation using sequencing techniques as \textit{ATAC-seq}, \textit{Sono-seq}, etc.
Even if there are some methods for the analysis of these omic data types, there still is a lack of them, in particular for emerging omics as \textit{ATAC-seq}.

To address this need, we decided to create a useful instrument for analysing chromatin regions accessibility data (such as \textit{ATAC-seq}, \textit{Sono-seq}).
Very often the biological questions, to be answered, as for \textit{RNA-seq}, need the comparison of two or more different biological conditions.
Starting from a set of already published \cite{Koberstein2018} scripts, we designed \gls{descan}, a software for the analysis of chromatin accession sequencing data.

In this chapter, we firstly illustrate the developed methodologies and then, with a case study, we will show the obtained results as an application.