To study multiple aspects of cellular behaviour related to different experimental conditions, due to drug treatments, pathologies, diseases, etc, several sequencing technologies have been developed during the last decades.

Starting with Sanger sequencing first and passing through Microarrays technologies then, nowadays with Next Generation Sequencing (NGS) we are able to understand many of biological mechanisms, like protein-chromatin interactions (e.g. ChIP-seq \cite{Park2009}), DNA methylation (Methyl-seq or BS-seq \cite{Frommer1992}), chromatin accessibility (e.g. ATAC-seq \cite{Buenrostro2013}), global transcriptional and translational activities (e.g. RNA-seq \ref{Wang2009}) and 3-D organization of chromatin (e.g. Hi-C \cite{VanBerkum2010}), giving the possibility to study same individual or experimental condition from many different points of view (transcriptomics, epigenomics, etc.) with a very high resolution. Each type of these “omics” data explains a different aspect of cellular behaviour. To give a comprehensive view of the cell regulatory mechanisms it is necessary not only to perform a single level analysis but also to provide novel statistical and computational models for integrating different omic types within a unified study.

Here we present the basics about the main sequencing techniques addressed in this thesis work, needed to better understand the outline of next chapters.
Please refer to the following chapters for details about the data analysis.