Multiple omics data information extraction became always more complex with the evolution of sequencing techniques and with the consequent development of methodologies for their analysis and integration.

With this thesis work, we focused on the development of novel strategies for several problems related to multiple omics sequencing data information extraction.
Firstly, we approached the most studied genomic problem, transcriptomic, by focusing on \gls{tc} \textit{RNA-seq} data.
We presented \textit{ticorser}, an R command line package for \gls{tc} \textit{RNA-seq} data.
With this instrument we propose a complete pipeline for the \textit{tc} data analysis, presenting also ad-hoc designed approaches for this type of data visualization at different levels.
Additionally, allowing a first integration level with functional annotation and also, in this case, their visualization.

Afterwards, because of the complexity of biological data, it has been almost mandatory to move our attention on epigenomics data, where we focused on \textit{ATAC-seq} technology, an emerging technique for open chromatin region detection, which still needs appropriate solutions for its analysis.
To address the lack of methodologies with this data we presented \textit{DEScan2}, an R/Bioconductor package allowing detection of open chromatin regions, and subsequently proposing a possible workflow for the detection of \glspl{der} and their integration with \textit{RNA-seq} data, allowing, moreover, a functional annotation integration.

Due to the difficulty of keeping track even of a single omic analysis, we presented \textit{easyReporting}, a possible approach to the Reproducible Computational Research inside R.
A tool which helps not only software developers to include \gls{rr} layers inside their tools, but also non-expert R analysts to easily produce reports for their analysis.

Finally, to speed up and help also non-expert users to analyse multiple omics data, we presented \gls{igro}, a web graphical user interface for the analysis of omics data.
Combining the flexibility of a point-and-click approach with the power of R/Bioconductor statistical approaches for the omics data analysis.
Moreover, thanks to the Reproducible Computational Research layer it allows to keep track of all the steps performed during the analysis, and with aid of dedicated interface to live edit the enriched report.